app/server/post_results.go

```diff
@@ -245,30 +250,48 @@ func writeToBlobStore(ctx context.Context, bucketURL, filename string, data []by
 	return nil
 }
 
-func extractAndVerifyCertForPayload(ctx context.Context, payload []byte) (*x509.Certificate, error) {
-	// Get most recent Rekor entry uuid.
-	uuids, err := getUUIDsByPayload(ctx, payload)
-	if err != nil || len(uuids) == 0 {
-		return nil, fmt.Errorf("error finding tlog entries corresponding to payload: %w", err)
+func extractAndVerifyCertForPayload(ctx context.Context, payload []byte, tlogIndex int64) (*x509.Certificate, error) {
+	var entry *tlogEntry
+	var uuid string
+	var err error
+
+	// #135 older versions of scorecard action wont send the tlog index, but newer ones will
+	if tlogIndex == noTlogIndex {
+		// Get most recent Rekor entry uuid.
+		uuids, err := getUUIDsByPayload(ctx, payload)
+		if err != nil || len(uuids) == 0 {
+			return nil, fmt.Errorf("error finding tlog entries corresponding to payload: %w", err)
+		}
+		uuid = uuids[len(uuids)-1] // ignore past entries.
+
+		// Get tlog entry from the UUID.
+		entry, err = getTLogEntryByUUID(ctx, uuid)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching tlog entry: %w", err)
+		}
+	} else {
+		uuid, entry, err = getTLogEntryByIndex(ctx, tlogIndex)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching tlog entry: %w", err)
+		}
 	}
-	// TODO(#135): We can't simply take the latest UUID. Either:
-	// (a) iterate through all returned UUIDs to find the right one.
-	// (b) send tlog index in the POST payload to identify the corresponding UUID.
-	uuid := uuids[len(uuids)-1] // ignore past entries.
 
-	// Get tlog entry from the UUID.
-	entry, err := getTLogEntry(ctx, uuid)
+	rekordBody, err := entry.rekord()
 	if err != nil {
-		return nil, fmt.Errorf("error fetching tlog entry: %w", err)
+		return nil, err
+	}
+
+	if !rekordBody.Matches(payload) {
+		return nil, errMismatchedTlogEntry
 	}
 
 	// Verify inclusion proof.
-	if err := verifyInclusionProof(uuid, entry); err != nil {
+	if err = verifyInclusionProof(uuid, entry); err != nil {
 		return nil, fmt.Errorf("verifying inclusion proof for Rekor: %w", err)
```

Comment:
nit
```suggestion
		return nil, fmt.Errorf("error verifying inclusion proof for Rekor: %w", err)
```

Reply:
I think error is a bit repetitive in an error message, but I can rephrase the original.

> unable to verify rekor inclusion proof

Reply:
Your version sounds good too.